src/engine/board.ts

```diff
@@ -49,13 +51,36 @@ export default class Board {
         return board;
     }
 
-    public squaresReachableInDirection(square: Square, direction : Offset){
+    private inBounds(square:Square) {
+        return square.row < gameSettings.BOARD_SIZE && square.col < gameSettings.BOARD_SIZE && square.row >= 0 && square.col >= 0
+    }
+
+    public offsetSquareAndCheckBounds(square:Square, offset:Offset):Square | null{
+        const offsetSquare = square.offset(offset);
+        if(this.inBounds(offsetSquare)){
+            return offsetSquare;
+        }
+        return null;
+    }
+
+    public enemyPieceIsOn(square : Square):boolean {
+        return this.getPiece(square)?.player != this.currentPlayer;
+    }
+
+    public squaresReachableInDirection(square: Square, offset : Offset){
         let squares:Square[] = [];
-        let nextSquare = square.squareAtOffset(direction);
+        let nextSquare = this.offsetSquareAndCheckBounds(square,offset);
         while(nextSquare && !this.getPiece(nextSquare)){
             squares.push(nextSquare);
-            nextSquare = nextSquare.squareAtOffset(direction);
+            nextSquare = this.offsetSquareAndCheckBounds(nextSquare,offset);
+        }
+
+        if(nextSquare && this.enemyPieceIsOn(nextSquare)){
```

Comment:
This is a subtle point, but one that's worth thinking about.

I think we're implicitly encoding the idea that pieces can take enemy pieces via their "normal" moves. In fact, this isn't true for all pieces (e.g. pawns), and it also doesn't feel like it's functionality associated with the board.